Service/EmigratieService.php

```diff
@@ -0,0 +1,201 @@
+    {
+        if (isset($zaakEigenschappen["MEEVERHUIZENDE_GEZINSLEDEN.MEEVERHUIZEND_GEZINSLID.BSN"])) {
+            return [
+                'MeeEmigrant' => [
+                    'emig:Burgerservicenummer' => $zaakEigenschappen["MEEVERHUIZENDE_GEZINSLEDEN.MEEVERHUIZEND_GEZINSLID.BSN"]
+                ]
+            ];
+        }// end if
+
+        $meeEmigranten = [];
+        $index = 1;
+        while (isset($zaakEigenschappen["MEEVERHUIZENDE_GEZINSLEDEN.MEEVERHUIZEND_GEZINSLID.$index.BSN"])) {
+            $meeEmigranten[] = [
+                'MeeEmigrant' => [
+                    'emig:Burgerservicenummer' => $zaakEigenschappen["MEEVERHUIZENDE_GEZINSLEDEN.MEEVERHUIZEND_GEZINSLID.$index.BSN"]
+                ]
+            ];
+            $index++;
+        }// end while
+
+        return $meeEmigranten;
+    } //end getMeeEmigranten()
+
+
+    /**
+     * This function gets the adressen from the zgwZaak with the given eigenschappen (simXml elementen and Stuf extraElementen).
+     *
+     * @param array  $zaakEigenschappen The zaak eigenschappen.
+     *
+     * @return array adressen
+     */
+    public function getAdressen(array $zaakEigenschappen): array
+    {
+        $adressen = [];
+        $index = 1;
+        while (isset($zaakEigenschappen["ADRESREGEL$index"])) {
+            $adressen["emig:AdresBuitenland$index"] = $zaakEigenschappen["ADRESREGEL$index"];
+            $index++;
+        }// end while
+
+        return $adressen;
+    } //end getAdressen()
+
+    /**
+     * Maps zgw eigenschappen to vrijbrp soap emigratie.
+     *
+     * @param ObjectEntity  $object The zgw case ObjectEntity.
+     * @param array $output The output data
+     *
+     * @return array
+     */
+    public function getEmigratieProperties(ObjectEntity $object, array $output): array
+    {
+        $this->logger->info('Do additional mapping with case properties');
+
+        $properties = ['all'];
+        $zaakEigenschappen = $this->zgwToVrijbrpService->getZaakEigenschappen($object, $properties);
+        $bsn = $this->zgwToVrijbrpService->getBsnFromRollen($object);
+        $output['soapenv:Body']['dien:AanvraagRequest']['dien:EmigratieaanvraagRequest']['emig:Aanvraaggegevens']['emig:BurgerservicenummerAanvrager'] = $bsn;
+        $output['soapenv:Body']['dien:AanvraagRequest']['dien:EmigratieaanvraagRequest']['emig:Aanvraaggegevens']['emig:Emigratiedatum'] = $zaakEigenschappen['DATUM_VERTREK'];
+        $output['soapenv:Body']['dien:AanvraagRequest']['dien:EmigratieaanvraagRequest']['emig:Aanvraaggegevens']['emig:LandcodeEmigratie'] = $zaakEigenschappen['LANDCODE'];
+        $output['soapenv:Body']['dien:AanvraagRequest']['dien:EmigratieaanvraagRequest']['emig:Aanvraaggegevens']['emig:AdresBuitenland'] = $this->getAdressen($zaakEigenschappen);
+        $output['soapenv:Body']['dien:AanvraagRequest']['dien:EmigratieaanvraagRequest']['emig:Aanvraaggegevens']['emig:MeeEmigranten'] = $this->getMeeEmigranten($zaakEigenschappen);
```

Comment:
Maybe do:
$output['soapenv:Body']['dien:AanvraagRequest']['dien:EmigratieaanvraagRequest']['emig:Aanvraaggegevens'] = [
'key' = 'value',
'key' = 'value',
]
instead?

Reply:
Writing this down for cleanup.

Reply:
Just done it for you in 30sec 👍